modal/_container_io_manager.py

```diff
@@ -662,13 +667,12 @@ async def _push_outputs(
             max_retries=None,  # Retry indefinitely, trying every 1s.
         )
 
-    def serialize_exception(self, exc: BaseException) -> Optional[bytes]:
+    def serialize_exception(self, exc: BaseException) -> bytes:
         try:
             return self.serialize(exc)
         except Exception as serialization_exc:
-            logger.info(f"Failed to serialize exception {exc}: {serialization_exc}")
             # We can't always serialize exceptions.
```

Comment:
Might want to keep the logging here, and in the serialization error print out `type(exc)` as well since the stringification of many exceptions is an empty string, e.g.:
```
>>> str(KeyboardInterrupt())
''
```